src/_P084_VEML6070.ino

```diff
@@ -0,0 +1,230 @@
+#define D_UV_INDEX_3  "High"     // = sun->glases a must
+#define D_UV_INDEX_4  "Danger"   // = sun->skin burns Level 1
+#define D_UV_INDEX_5  "BurnL1/2" // = sun->skin burns level 1..2
+#define D_UV_INDEX_6  "BurnL3"   // = sun->skin burns with level 3
+#define D_UV_INDEX_7  "OoR"      // = out of range or unknown
+
+#include <math.h>  
+
+double     uv_risk_map[VEML6070_UV_MAX_INDEX] = {0,0,0,0,0,0,0,0,0,0,0,0,0,0,0};
+char       str_uvrisk_text[10];
+
+boolean Plugin_084(byte function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+      {
+        Device[++deviceCount].Number = PLUGIN_ID_084;
+        Device[deviceCount].Type = DEVICE_TYPE_I2C;
+        Device[deviceCount].Ports = 0;
+        Device[deviceCount].VType = SENSOR_TYPE_SINGLE;
+        Device[deviceCount].PullUpOption = false;
+        Device[deviceCount].InverseLogicOption = false;
+        Device[deviceCount].FormulaOption = true;
+        Device[deviceCount].ValueCount = 3;
+        Device[deviceCount].SendDataOption = true;
+        Device[deviceCount].TimerOption = true;
+        Device[deviceCount].TimerOptional = false;
+        Device[deviceCount].GlobalSyncOption = true;
+        break;
+      }
+
+    case PLUGIN_GET_DEVICENAME:
+      {
+        string = F(PLUGIN_NAME_084);
+        break;
+      }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+      {
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_084));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_084));
+        strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_084));
+        break;
+      }
+
+    case PLUGIN_WEBFORM_LOAD:
+      {
+
+        String optionsMode[4] = { F("1/2T"), F("1T"), F("2T"), F("4T (Default)") };
+        addFormSelector(F("Refresh Time Determination"), F("itime"), 4, optionsMode, NULL, PCONFIG(1));
+
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_SAVE:
+      {
+        //PCONFIG(0) = getFormItemInt(F("i2c_addr"));
+        PCONFIG(1) = getFormItemInt(F("itime"));
+
+        success = true;
+        break;
+      }
+
+    case PLUGIN_INIT:
+      {
+        bool error = VEML6070_Init(PCONFIG(1));
+        if(error){
+          String log = F("VEML6070: Not available!");
+          addLog(LOG_LEVEL_INFO, log);
+        }
+        
+        VEML6070_UvTableInit();
+
+        success = true;
+        break;
+      }
+
+    case PLUGIN_READ:
+      {
+        uint16_t uv_raw;
+        double uv_risk, uv_power;
+
+        uv_raw   = VEML6070_ReadUv();             // get UV raw values
+        uv_risk  = VEML6070_UvRiskLevel(uv_raw);  // get UV risk level
+        uv_power = VEML6070_UvPower(uv_risk);     // get UV power in W/m2
+
+        if (isnan(uv_raw) || uv_raw == (-1) || uv_raw == 65535) {
+          String log = F("VEML6070: no data read!");
+          addLog(LOG_LEVEL_INFO, log);
```

Comment:
`addLog(LOG_LEVEL_INFO, F("VEML6070: no data read!"));` prevents string allocation if the loglevel is not active.

Reply:
Code changed to prevent string allocation if the loglevel is not active.